### PR TITLE
Fixes #21067: Add windows support to the inventory software update section

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/21067

### CODE REVIEW

File: rudder-agent/SOURCES/patches/fusioninventory/21067-windows-updates.patch [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
++        print $fh $raw_json;
++        print $fh "\n";
++        print $fh $@;
++        close($fh);
++        $logger->error("Invalid JSON data stored in $filename") if $logger;
++    }
++    foreach my $update (@updates) {
++      my $package = {
++        NAME    => $update->{Name},
++        ARCH    => "x86_64",
```
Comment (amousset) [inline rudder-agent/SOURCES/patches/fusioninventory/21067-windows-updates.patch:97]: this needs to be replaced by the real value

File: rudder-agent/SOURCES/patches/fusioninventory/21067-windows-updates.patch [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
++        VERSION => "1.0",
```
Comment (amousset) [inline rudder-agent/SOURCES/patches/fusioninventory/21067-windows-updates.patch:98]: if we don't have a version number we need to make it optional on the webapp side

File: rudder-agent/SOURCES/patches/fusioninventory/21067-windows-updates.patch [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
++        open(my $fh, '>', $filename);
++        print $fh $raw_json;
++        print $fh "\n";
++        print $fh $@;
++        close($fh);
++        $logger->error("Invalid JSON data stored in $filename") if $logger;
++    }
++    foreach my $update (@updates) {
++      my $package = {
++        NAME    => $update->{Name},
```
Comment (amousset) [inline rudder-agent/SOURCES/patches/fusioninventory/21067-windows-updates.patch:96]: I think we should use the KB id here to match what is done for installed packages.

We could reuse the string concatenated with the KB description in the DESCRIPTION field.

**Comment by Fdall** [comment]
Commit modified

**Pull request merged**